From 39880eb7feeac561a2b1e91eb79657380238a6f2 Mon Sep 17 00:00:00 2001
From: Arlo Phoenix <aarlo.phoenix@gmail.com>
Date: Wed, 7 Aug 2024 15:42:48 +0200
Subject: [PATCH 3/9] Default to building for all architectures

that build command is pretty long so just reuse
PYTORCH_ROCM_ARCH and pass it to CMAKE_HIP_ARCHITECTURES
---
 CMakeLists.txt | 3 ++-
 README_ROCM.md | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 645717d9..94a5bcfa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -583,7 +583,8 @@ if (WITH_HIP)
   enable_language(HIP)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   message(STATUS "HIP Compiler: ${CMAKE_HIP_COMPILER}")
-  #add_definitions(-D_GLIBCXX_USE_CXX11_ABI=0)
+  message(STATUS "CMAKE_HIP_ARCHITECTURES: ${CMAKE_HIP_ARCHITECTURES}")
+  
   if(NOT DEFINED ENV{ROCM_PATH})
     set(ROCM_PATH /opt/rocm)
   else()
diff --git a/README_ROCM.md b/README_ROCM.md
index d9fdc994..d7b4cfe6 100644
--- a/README_ROCM.md
+++ b/README_ROCM.md
@@ -16,7 +16,8 @@ conda activate py_3.9
 ```bash
 git clone https://github.com/arlo-phoenix/CTranslate2-rocm.git --recurse-submodules
 cd CTranslate2-rocm
-CLANG_CMAKE_CXX_COMPILER=clang++ CXX=clang++ HIPCXX="$(hipconfig -l)/clang" HIP_PATH="$(hipconfig -R)"     cmake -S . -B build -DWITH_MKL=OFF -DWITH_HIP=ON -DCMAKE_HIP_ARCHITECTURES=gfx1030 -DBUILD_TESTS=ON -DWITH_CUDNN=ON
+#export PYTORCH_ROCM_ARCH=gfx1030 #optionally set this only to your ROCm arch to speed up compiling. You can find it with rocminfo | grep gfx
+CLANG_CMAKE_CXX_COMPILER=clang++ CXX=clang++ HIPCXX="$(hipconfig -l)/clang" HIP_PATH="$(hipconfig -R)"     cmake -S . -B build -DWITH_MKL=OFF -DWITH_HIP=ON -DCMAKE_HIP_ARCHITECTURES=$PYTORCH_ROCM_ARCH -DBUILD_TESTS=ON -DWITH_CUDNN=ON
 cmake --build build -- -j16
 cd build
 cmake --install . --prefix $CONDA_PREFIX #or just sudo make install if not using conda env
-- 
2.47.1

